pkg/cmd/server/origin/master.go

```diff
@@ -609,8 +609,8 @@ func (c *MasterConfig) RunBuildImageChangeTriggerController() {
 func (c *MasterConfig) RunDeploymentController() {
 	_, kclient := c.DeploymentControllerClients()
 	env := []api.EnvVar{
-		{Name: "KUBERNETES_MASTER", Value: c.MasterAddr},
-		{Name: "OPENSHIFT_MASTER", Value: c.MasterAddr},
+		{Name: "KUBERNETES_MASTER", Value: c.MasterPublicAddr},
```

Comment:
Doesn't that mean --master is set incorrectly? `--master` should be the address in-cluster things (e.g. nodes) can use to reach the master. `--public-master` is the address out of cluster things (e.g. browser, `osc` from user's desktop) can use to reach the master.